omc3/global_correction.py

```diff
@@ -0,0 +1,356 @@
+
+**Arguments:**
+
+*--Required--*
+
+- **meas_dir**:
+
+    Path to the directory containing the measurement files.
+
+
+- **output_dir**:
+
+    Path to the directory where to write the output files.
+
+
+*--Optional--*
+
+- **beta_file_name**:
+
+    Prefix of the beta file to use. E.g.: getkmodbeta
+
+    default: ``beta_phase_``
+
+
+- **errorcut** *(float)*:
+
+    Reject BPMs whose error bar is higher than the corresponding input.
+    Input in order of optics_params.
+
+
+- **fullresponse_path**:
+
+    Path to the fullresponse binary file.If not given, calculates the
+    response analytically.
+
+
+- **max_iter** *(int)*:
+
+    Maximum number of correction re-iterations to perform. A value of `0`
+    means the correction is calculated once.
+
+    default: ``3``
+
+
+- **method** *(str)*:
+
+    Optimization method to use.
+
+    choices: ``('pinv', 'omp')``
+
+    default: ``pinv``
+
+
+- **min_corrector_strength** *(float)*:
+
+    Minimum (absolute) strength of correctors.
+
+    default: ``0.0``
+
+
+- **modelcut** *(float)*:
+
+    Reject BPMs whose deviation to the model is higher than the
+    correspoding input. Input in order of optics_params.
+
+
+- **n_correctors** *(int)*:
+
+    Maximum number of correctors to use. (Method: 'omp')
+
+
+- **optics_params** *(str)*:
+
+    List of parameters to correct upon (e.g. BETX BETY)
+
+    choices: ``('PHASEX', 'PHASEY', 'BBX', 'BBY', 'BETX', 'BETY', 'DX', 'DY', 'NDX', 'Q', 'F1001R', 'F1001I', 'F1010R', 'F1010I')``
+
+    default: ``['PHASEX', 'PHASEY', 'BETX', 'BETY', 'NDX', 'Q']``
+
+
+- **output_filename**:
+
+    Identifier of the output files.
+
+    default: ``changeparameters_iter``
+
+
+- **svd_cut** *(float)*:
+
+    Cutoff for small singular values of the pseudo inverse. (Method:
+    'pinv')Singular values smaller than rcond*largest_singular_value are
+    set to zero
+
+    default: ``0.01``
+
+
+- **update_response**:
+
+    Update the (analytical) response per iteration.
+
+    action: ``store_true``
+
+
+- **use_errorbars**:
+
+    Take into account the measured errorbars in the correction.
+
+    action: ``store_true``
+
+
+- **variable_categories**:
+
+    List of names of the variables classes to use.
+
+    default: ``['MQM', 'MQT', 'MQTL', 'MQY']``
+
+
+- **weights** *(float)*:
+
+    Weight to apply to each measured quantity. Input in order of
+    optics_params.
+
+
+
+Possible problems and notes (lmalina, 2020):
+ * error-based weights default? likely - but be careful with low tune errors vs svd cut in pseudoinverse
+ * manual creation of pd.DataFrame varslist, deltas? maybe tunes in tfs_pandas single value or a column?
+ * There should be some summation/renaming for iterations
+ * For two beam correction
+ * The two beams can be treated separately until the calculation of correction
+ * Values missing in the response (i.e. correctors of the other beam) shall be treated as zeros
+ * Missing a part that treats the output from LSA
```

Comment:
There are also these comments from Lukas, where I am mostly not sure about what exactly they mean. Could be transformed into issues?

Reply:
To me these things are not even issues but rather possible improvements. I think these can just be put in one issue labeled improvement for now.